File: spec/python/test_imports_abs_abs.py
import unittest

from imports_abs_abs import ImportsAbsAbs

class TestImportsAbsAbs(unittest.TestCase):
    def test_imports_abs_abs(self):
        r = ImportsAbsAbs.from_file("src/fixed_struct.bin")

        self.assertEqual(r.one, 0x50)
        self.assertEqual(r.two.one, 0x41)
        self.assertEqual(r.two.two.one, 0x43)

File: spec/lua/test_imports_abs_abs.lua
local luaunit = require("luaunit")

require("imports_abs_abs")

TestImportsAbsAbs = {}

function TestImportsAbsAbs:test_imports_abs_abs()
    local r = ImportsAbsAbs:from_file("src/fixed_struct.bin")

    luaunit.assertEquals(r.one, 0x50)
    luaunit.assertEquals(r.two.one, 0x41)
    luaunit.assertEquals(r.two.two.one, 0x43)
end
